porcupine/tabs.py

```diff
@@ -598,7 +599,7 @@ def __init__(self, manager: TabManager, content: str = "", path: Path | None = N
         )
         self.settings.add_option("tabs2spaces", True)
         self.settings.add_option("indent_size", 4)
-        self.settings.add_option("encoding", "utf-8")
+        self.settings.add_option("encoding", self._detect_encoding())
```

Comment:
I think we don't need to call `_detect_encoding()` here, because it will be called on the first reload anyway. (The `open_file()` method calls `reload()`)

Another thought: I found this comment, which made me think it should be possible to make the BOM detection a plugin.

```python3
        # Add tab before loading content, so that editorconfig plugin gets a
        # chance to set the encoding into tab.settings
```
That can be a separate PR: this PR is very good as is :)

Reply:
If we keep this line, `"utf-8"` should still probably be the default value. I think it is currently not possible to reset a setting to its default, but changing the default value of the setting might have weird consequences now or in a future version of Porcupine.

```suggestion
        self.settings.add_option("encoding", "utf-8")
        self.settings.set("encoding", self._detect_encoding())
```